ui/src/common/components/TreeAggregation.jsx

```diff
@@ -0,0 +1,164 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import Immutable from 'immutable';
+import { Col, Row, Tree } from 'antd';
+
+import UnclickableTag from './UnclickableTag';
+import AggregationBox from './AggregationBox';
+import { forceArray } from '../utils';
+import FormattedNumber from './FormattedNumber';
+
+export const BUCKET_NAME_SPLITTER = '_';
+
+class TreeAggregation extends Component {
```

Comment:
we don't write new components this way anymore, instead prefer function components with hooks.

Reply:
you can start writing the tests, then refactor which should make it easier.